ledger/src/shred.rs

```diff
@@ -290,16 +290,16 @@ impl Shred {
 
     pub fn new_from_serialized_shred(mut payload: Vec<u8>) -> Result<Self, Error> {
         let mut cursor = Cursor::new(&payload[..]);
-        let common_header: ShredCommonHeader = bincode::deserialize_from(&mut cursor)?;
+        let common_header: ShredCommonHeader = deserialize_from_with_limit(&mut cursor)?;
         let (data_header, coding_header) = match common_header.shred_type {
             ShredType::Code => {
-                let coding_header = bincode::deserialize_from(&mut cursor)?;
```

Comment:
Looks like the older version of [bincode::deserialize_from](https://docs.rs/bincode/0.8.0/bincode/fn.deserialize_from.html#:~:text=Function%20bincode%3A%3Adeserialize_from&text=Deserializes%20an%20object%20directly%20from,your%20server%20out%20of%20memory.) actually includes a size limit while the newer version does not.  Do you happen to know why the API evolves in this way?

Reply:
not really